packages/react/src/components/FEaaS/feaas-utils.ts

```diff
@@ -110,12 +122,13 @@ async function fetchData(dataOptions: FEAAS.DataOptions): Promise<FEAAS.DataScop
  * @returns component endpoint URL
  */
 export const composeComponentEndpoint = (
-  params: FEaaSComponentParams,
+  params: FEaaSComponentParams & RequiredFEaaSParams,
   revisionFallback: RevisionType
 ) => {
   const revision = params.ComponentRevision || revisionFallback;
   const hostname = params.ComponentHostName.startsWith('https://')
     ? params.ComponentHostName
     : `https://${params.ComponentHostName}`;
```

Comment:
For simplicity you can remove `RequiredFEaaSParams`, `isFEaaSComponentParamsComplete`, `ValidatedFEaaSComponentParams` things and just add optional chaining for `ComponentHostName`
```suggestion
  params: FEaaSComponentParams,
  revisionFallback: RevisionType
) => {
  const revision = params.ComponentRevision || revisionFallback;
  const hostname = params.ComponentHostName?.startsWith('https://')
    ? params.ComponentHostName
    : `https://${params.ComponentHostName}`;
```